src/request.ts

```diff
@@ -103,29 +111,38 @@ export const parseError = (
 };
 
 /**
- * Designed to work with `checkStatus`, or any function that
+ * Designed to work with `_checkStatus`, or any function that
  * raises an error on an invalid status. The error raised should have a `response`
  * property with the original response object.
  *
  * Example usage:
  *
- * fetchJSON('/api/v3/test/path', {'body': someData})
- *     .catch(catchStatusError)
- *     .catch(({response, parsedError}) => doSomethingOnError())
- *     .then(doSomethingOnSuccess);
+ * _fetchJSON('/api/v3/test/path', {'body': someData})
+ *     .catch(_catchStatusError)
+ *     .then(doSomethingOnSuccess)
+ *     .catch(({response, parsedError}) => doSomethingOnError());
  */
-export const catchStatusError = (response: Response): Promise<any> =>
+const _catchStatusError = (res: Response): Promise<any> =>
     new Promise((resolve, reject) => {
-        response
-            .json()
```

Comment:
was the issue that the actual parsing was failing and overriding the potential error from the api?

Reply:
Yeah if for whatever reason we weren't getting back valid JSON, this was failing. It was one of the new test cases I added. Our actual code was failing instead of us providing a rejected promise